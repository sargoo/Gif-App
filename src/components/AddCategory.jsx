import PropTypes  from "prop-types";
import React, { useState } from "react";

export const AddCategory = ({addCategory}) =>
{
    const  [inputValue, setInputValue] = useState ('');

    const handleInputValue = (e) =>{
        
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        if(inputValue.trim().length < 1) return;
        addCategory(inputValue.trim());
        setInputValue('');

    }



    return(
        <>
            <form onSubmit={handleSubmit} aria-label = 'form'>
            <input  type="text" 
                    placeholder="Buscar Gif"
                    value={inputValue}
                    onChange ={(e) => {handleInputValue(e)}}
                />
            </form>
        </>
    );
}

AddCategory.propTypes = {
    addCategory: PropTypes.func.isRequired,
};