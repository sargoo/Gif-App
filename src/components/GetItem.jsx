import PropTypes  from 'prop-types';

export const GetItem = ({url, title, id, handleFilterGif}) =>{

    return (
        <div className="card">
            <img src={url} alt="gif"></img>
            <p data-testid = 'test-title'> {title}</p>
            <div className="boton">
                <button onClick={handleFilterGif} id={id}> Quitar gif</button>
            </div>
        </div>
    );
}

GetItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}