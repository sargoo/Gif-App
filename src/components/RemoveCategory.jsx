export const RemoveCategory = ({category, handleFilterCategory} ) =>{
   
    return(
        <>
            <div className="boton">
                <button onClick={handleFilterCategory} id={category}>Quitar {category} </button>
            </div>
        </>
    );
}