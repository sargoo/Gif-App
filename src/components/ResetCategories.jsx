export const ResetCategories = ({handleReset, categories}) =>{

    return(
        <>
            <div className="boton">
                <button onClick={handleReset} id = {categories}> RESET</button>
            </div>
        </>
    );
}