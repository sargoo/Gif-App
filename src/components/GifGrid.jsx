import { useFetchGif } from "../hooks/UseFetchGif";
import { GetItem } from "./GetItem";
import { RemoveCategory } from "./RemoveCategory";
import  PropTypes  from 'prop-types';


export const GifGrid = ({category}) =>{

    const handleFilterCategory = (e) =>{
        setGifs(gifs.filter(gif => gif.category !== e.target.id));        
    }

    const handleFilterGif = (e) =>{
        setGifs(gifs.filter(gif => gif.id !== e.target.id));
    }

    const {gifs, loading, setGifs} = useFetchGif(category);

    
    return(
        <>
             
            <h3> {category}</h3> 
            <RemoveCategory category={category} handleFilterCategory={handleFilterCategory}/>
            <div className="card-grid">
            {
                loading? <h2> Cargando...</h2> : gifs.map( gif => (<GetItem  key={gif.id} {...gif} handleFilterGif ={handleFilterGif}/>))
            }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

