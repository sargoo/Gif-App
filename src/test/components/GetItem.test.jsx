import { render,screen } from "@testing-library/react";
import { GetItem } from "../../components/GetItem";

describe('Test en <GetItem />', () => {

    const title = 'Star Wars';
    const url = 'http://www.starwar.com/starwar.jpg';
    
    // test('Debe coincidir con el snapshot', () => {

    //     const { container } = render(<GetItem title={title} url={url}/>);

    //     expect(container).toMatchSnapshot();

    // });

    test('Debe encontrar un texto', () => {

        const {getByText} = render(<GetItem title={title} url={url}/>);

        expect(getByText(title)).toBeTruthy();

    });

    test('Debe encontrar por test-id', () => {
        
        const {getByTestId} = render(<GetItem title={title} url={url}/>);

        expect(getByTestId('test-title').innerHTML).toContain(title);

     });

     test('Debe mostrar URL y ALT', () => {

        render(<GetItem title={title} url={url}/>); 
        
        //screen.debug();
        


        const {src,alt} = screen.getByRole('img');
        expect(src).toContain(url);
        expect(alt).toContain('gif');
        

     });
});