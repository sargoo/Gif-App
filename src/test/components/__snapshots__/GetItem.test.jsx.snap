// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GetItem /> Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="gif"
      src="http://www.starwar.com/starwar.jpg"
    />
    <p
      data-testid="test-title"
    >
       
      Star Wars
    </p>
    <div
      class="boton"
    >
      <button>
         Quitar gif
      </button>
    </div>
  </div>
</div>
`;
